paddle/fluid/pir/dialect/operator/ir/ops.yaml

```
@@ -1420,6 +1420,16 @@
   optional: dropout1_seed, dropout2_seed, linear1_bias, linear2_bias, ln1_scale, ln1_bias, ln2_scale, ln2_bias, ln2_mean, ln2_variance, ln1_mean, ln1_variance, ln1_out
   backward: fused_feedforward_grad
 
+- op: retinanet_detection_output
+  args: (Tensor[] bboxes, Tensor[] scores, Tensor[] anchors, Tensor iminfo, float score_threshold, int64_t nms_top_k, float nms_threshold, float nms_eta, int64_t keep_top_k)
+  output: Tensor(out)
+  infer_meta:
+    func: RetinanetDetectionOutputInferMeta
+    param: [bboxes, scores, anchors, iminfo]
+  kernel:
+    func: retinanet_detection_output
+  optional: bboxes, scores, anchors
```

```suggestion
- op: retinanet_detection_output
  args: (Tensor[] bboxes, Tensor[] scores, Tensor[] anchors, Tensor iminfo, float score_threshold, int64_t nms_top_k, float nms_threshold, float nms_eta, int64_t keep_top_k)
  output: Tensor(out)
  infer_meta:
    func: RetinanetDetectionOutputInferMeta
    param: [bboxes, scores, anchors, iminfo]
  kernel:
    func: retinanet_detection_output
    data_type : scores
  interfaces: RetinanetDetectionOutputOpParseKernelKey
```
这里有两个问题跟你确认下：
1. 为什么要添加optional字段呢？我看retinanet_detection_output_op.cc里面对这些输入声明了AsDuplicable，但是并没有声明AsDispensable
2. interfaces字段的添加，这个是为了迁移原算子体系下RetinanetDetectionOutputOp::GetExpectedKernelType引入的。除了在yaml配置字段，还需要在 `paddle/fluid/pir/dialect/operator/interface/interface.cc`里定义一个RetinanetDetectionOutputOpParseKernelKey函数，可以参考 #59124 定义的UniqueOpParseKernelKey。